src/zope/i18nmessageid/message.py

```diff
@@ -37,37 +37,37 @@ def __new__(cls, ustr, domain=_marker, default=_marker, mapping=_marker,
                 msgid_plural=_marker, default_plural=_marker, number=_marker):
         self = six.text_type.__new__(cls, ustr)
         if isinstance(ustr, self.__class__):
-            domain = ustr.domain[:] if domain is _marker else domain
-            default = ustr.default[:] if default is _marker else default
-            mapping = ustr.mapping.copy() if mapping is _marker else mapping
-            msgid_plural = (
-                ustr.msgid_plural[:] if msgid_plural is _marker else
-                msgid_plural)
-            default_plural = (
-                ustr.default_plural[:] if default_plural is _marker else
-                default_plural)
-            number = ustr.number if number is _marker else number
-            ustr = six.text_type(ustr)
+            self.domain = ustr.domain[:] if ustr.domain else None
+            self.default = ustr.default[:] if ustr.default else None
+            self.mapping = ustr.mapping.copy() if ustr.mapping is not None else None
+            self.msgid_plural = ustr.msgid_plural[:] if ustr.msgid_plural else None
+            self.default_plural = ustr.default_plural[:] if ustr.default_plural else None
```

Comment:
Why are we even trying to slice strings anyway?  `a_string[:] is a_string` in CPython AFAICT.  Is the C code doing slices?

(Also I'm mildly unhappy about explicitly making a copy of the mapping dictionary and then throwing it away.  I suppose it's a rare occurrence, but still.)

Reply:
I wondered that too. The C code is *not* slicing strings. Nor does it `copy()` the mapping. Suppose we just remove that here?

Reply:
No tests broke if that stops happening. I removed it and added tests to confirm that Python and C  are handling the objects the same way.